tests/test_target.py

```diff
@@ -0,0 +1,122 @@
+"""
+Unit-tests for target generation
+"""
+
+import click
+import pytest
+import commodore.cluster as cluster
+
+
+cluster_obj = {
+    'id': 'mycluster',
+    'tenant': 'mytenant',
+    'facts': {
+        'distribution': 'rancher',
+        'cloud': 'cloudscale',
+    }
+}
+components = ['test-component']
+catalog = 'ssh://git@git.example.com/cluster-catalogs/mycluster'
+
+
+def test_render_target():
+    target = cluster._full_target(cluster_obj, components, catalog)
+    facts = cluster_obj['facts']
+    assert target != ""
+    all_classes = ([f"defaults.{cn}" for cn in components] +
+                   ['global.common',
+                       f"global.{facts['distribution']}",
+                       f"global.{facts['cloud']}",
+                       f"{cluster_obj['tenant']}.{cluster_obj['id']}",
+                    ])
+    assert len(target['classes']) == len(
+        all_classes), "rendered target includes different amount of classes"
```

Comment:
I'd suggest this style: (of course, personal taste)

```suggestion
    assert len(target['classes']) == len(all_classes), \
        "rendered target includes different amount of classes"
```

If you write it like this the actual test is on a single line. Note that the text afterwards, which is _totally optional_, is merely a comment and helps to understand the error in case the test fails.

Reply:
Agree, autopep made this kinda uggly

Reply:
See https://github.com/projectsyn/commodore/pull/89 for implementation of the findings